test/checkChallenge.test.js

```diff
@@ -37,68 +44,1880 @@ describe('Challenges', () => {
+  
+      config.minmaintimelive = 352;
+      config.maxmaintimelive = 4127;
+      
+      const result = conn.checkChallengeTimeSettings(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Minimum Total Time for live games in absolute is 5 minutes 52 seconds, please increase Total Time.' }));
+    });
+  
+    it('accept main time (total time) edge min', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "absolute", time_control: "absolute", speed: "live", total_time: 352 } });
+  
+      config.minmaintimelive = 352;
+      config.maxmaintimelive = 4127;
+      
+      const result = conn.checkChallengeTimeSettings(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+  
+    it('accept main time (total time) between min and max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "absolute", time_control: "absolute", speed: "live", total_time: 1249 } });
+  
+      config.minmaintimelive = 352;
+      config.maxmaintimelive = 4127;
+      
+      const result = conn.checkChallengeTimeSettings(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+  
+    it('accept main time (total time) edge max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "absolute", time_control: "absolute", speed: "live", total_time: 4127 } });
+  
+      config.minmaintimelive = 352;
+      config.maxmaintimelive = 4127;
+  
+      const result = conn.checkChallengeTimeSettings(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+  
+    it('reject main time (total time) too high', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "absolute", time_control: "absolute", speed: "live", total_time: 4128 } });
+  
+      config.minmaintimelive = 352;
+      config.maxmaintimelive = 4127;
+      
+      const result = conn.checkChallengeTimeSettings(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum Total Time for live games in absolute is 1 hours 8 minutes 47 seconds, please reduce Total Time.' }));
+    });
+
+    // Periods are not checked for non-byoyomi time controls
+
+    // Period Time is not checked for absolute time control
+
+  });
+
+  describe('Min Max General Ranked Unranked precdecence rules', () => {
+
+    // We already tested extensively how the min max args work, so now we just want to
+    // make sure the general / ranked / unranked priority order is respected.
+    // MinMax handicap is a good and simple example
+
+    it('reject handicap based on ranked arg if ranked arg is used and game is ranked', () => {
+
+      const notification = base_challenge({ ranked: true, handicap: 8 });
+
+      config.noautohandicap         = true;
```

Comment:
should we also remove all the extra spaces formatting here too?

Reply:
Yes